src/types/options.ts

```diff
@@ -74,6 +74,7 @@ export type Typescript = {
   tsconfigFile?: string | boolean;
   tsconfigDirectory?: string | boolean;
   reportDiagnostics?: boolean;
+  useAdvancedImportTranspiler?: boolean;
```

Comment:
Don't know if this is the best name for the option, better names welcome

Reply:
Do we really need a new option? This new process should impact existing codebases.

Reply:
My arguments in favor of the option:
- backwards compatible. If this is the new default, then either it's a breaking change because people would need to install sorcery and magic-string, or we add them to `dependencies` but then everyone, even those who don't use TypeScript, need to download the additional packages
- intermediate solution: there are plans from the TypeScript team to provide a new import option which would strictly separate types and values, which would make this advanced transpilation unnecessary. If that solution is implemented, people are free to choose the solution right here or use the new TypeScript option.

Reply:
FWIW adding both as dependencies adds +- 1.6mb to the node_modules (ignoring possible dep deduplication).

Reply:
I like the idea of having an intermediate solution. If there's a possibility that this can be solved in a future ts version, we should plan this change with that in mind. If we go through with this, should we say anything about it being a temporary solution?

Reply:
Good idea about adding a note. I'm not 100% sure if/when the TS solution lands, so the note should reflect that. And even after that, people might prefer not strictly separating types and values and still want to choose this "temporary" solution.